src/ComponentHelper.js

```diff
@@ -1,25 +1,40 @@
 var deepEqual = require('deep-equal');
+const cameraWorldPosition = new THREE.Vector3();
+const cameraGroupConjugate = new THREE.Quaternion();
 
 module.exports.gatherComponentsData = function(el, schemaComponents) {
-  var elComponents = el.components;
   var compsData = {};
 
   for (var i in schemaComponents) {
     var element = schemaComponents[i];
 
     if (typeof element === 'string') {
-      if (elComponents.hasOwnProperty(element)) {
+      if (el.components.hasOwnProperty(element)) {
         var name = element;
-        var elComponent = elComponents[name];
-        compsData[name] = AFRAME.utils.clone(elComponent.data);
+        // In VR mode, aframe gives over control of the camera to three's WebVRManager. WebVRManager modifies
+        // the camera's matrixWorld directly, so we have to decompose the camera's position ourselves.
+        // However, the camera's world position incorporates the camera's rotation in a weird way, so we have to
+        // undo that rotation as well.
+        if (name === 'position' && el.components.hasOwnProperty('camera') && el.sceneEl.is('vr-mode')) {
+          el.components.camera.camera.getWorldPosition(cameraWorldPosition);
+          cameraGroupConjugate.copy(el.object3D.quaternion);
+          cameraGroupConjugate.conjugate();
+          cameraWorldPosition.sub(el.object3D.position);
+          cameraWorldPosition.applyQuaternion(cameraGroupConjugate);
+          cameraWorldPosition.add(el.object3D.position);
+          compsData[name] = AFRAME.utils.clone(cameraWorldPosition);
```

Comment:
This feels overly complicated. I'm going to find out if this is a mistake with the underlying code, but this works around things for now.

Reply:
Maybe we can add a method to the camera component in AFrame core to get its position vs doing it in libraries?

Reply:
Hmm yeah that fact that you would have to do this in a library feels very wrong. if the position/rotation can't be updated directly maybe this should be a special case handled in the position and rotation component's themselves.

Reply:
@dmarcos ^ for reference, this is what I have to do to get the camera's actual position now.